encode.go

```diff
@@ -406,7 +408,12 @@ func (e *Encoder) isNeedQuoted(v string) bool {
 
 func (e *Encoder) encodeString(v string, column int) ast.Node {
 	if e.isNeedQuoted(v) {
-		v = strconv.Quote(v)
+		switch e.quoteStyle {
+		case QuoteStyleSingle:
+			v = quoteWith(v, '\'')
```

Comment:
This is difficult. In the original Issue, the submitter worried that originally defined string in a single quote changes to double quoted string at marshal. In this case, I think it is enough to always be able to select single quote or double quote for "quoted string" ( e.g. contains special character like `{` or `}` ) .

For example, when outputting a character containing a new line code ( `\n` or `\r` ), I think that using `Literal` ( `|` ) is more readable than a quoted character string.

Reply:
Ok, yeah:
1. I'll mark this PR as not actually closing the linked issue. It solves a different issue: I wanted to be able to turn all double quoted strings in the output into single quoted strings, so definitely not the same as the issue I referenced.
2. There's a much bigger discussion to be had about how quoting should behave. It seems like right now, the quoting is not very smart and it needs to get much fancier. I'd love to chat about this synchronously rather than over github comments because the discussion would take forever this way. Let me know your preferred method of communication. You can DM me on twitter at @vikstrous

Reply:
OK ! I would like to discuss on `gophers.slack.com` .  I created `#goccy-go-yaml` channel .
Could you join it ?